specs/LoaderDllFolderPath.md

```diff
@@ -0,0 +1,47 @@
+# Background
+The WebView2 loader code is what knows where to find and start or connect to already running WebView2 runtimes. This can either be statically linked for C based projects, or is also available as a standalone DLL. For .NET projects they must use the DLL since the loader code is native and we don't have a way to merge the native code into a managed module. Before this change the .NET WebView2 API module would look for the WebView2Loader.dll in the its same folder. But some .NET projects have requirements about where they can place DLLs and so this property is introduced to allow end developers to place the WebView2Loader.dll in any folder and specify the path of 'WebView2Loader.dll' explicitly.
+So we add an API to enable them to specify the path, so that they can use 'WebView2Loader.dll' from any path.
+
+# Description
+We add new static `LoaderDllFolderPath` property in `CoreWebView2Environment` class.
+The property allow the end developers to specify the folder's path containing `WebView2Loader.dll`.
+
+# Examples
+## C#
+```c#
+Task<CoreWebView2Environment> CreateEnvironmentAsync()
+{
+    // Specify a relative path '.'. There should be a 'WebView2Loader.dll' file in the folder.
+    CoreWebView2Environment.LoaderDllFolderPath = ".";
+    return CoreWebView2Environment.CreateAsync();
+}
+```
+
+# Remarks
+This property allows you to set the path of the folder containing the `WebView2Loader.dll`. This should be the path of a folder containing `WebView2Loader.dll` and not a path to the `WebView2Loader.dll` file itself.
+Note that the WebView2 SDK contains multiple `WebView2Loader.dll` files for different CPU architectures. When specifying folder path, you must specify one containing a `WebView2Loader.dll` module with a CPU architecture matching the current process CPU architecture.
+This property is used to load the `WebView2Loader.dll` module during calls to any of the static methods on `CoreWebView2Environment`. So, the path should be specified before any API called in `CoreWebView2Environment` class. Once `WebView2Loader.dll` is successfully loaded this property is no longer used.
```

Comment:
Okay, I think I see why this is a static property.

Originally, I thought a better place would be on `CoreWebview2EnvironmentOptions`. That removes the question of "When is it safe to change it?"

Putting it in the CoreWebview2EnvironmentOptions makes it clear that
* You must decide before you do anything with the CoreWebview2Environment
* Any decision you make is locked in for the lifetime of the environment, just like BrowserVersionString and UserDataFolder.

Unfortunately, that doesn't work due the existence of `GetAvailableBrowserVersionString()`, which is a static method but also needs to locate the WebView2 runtime.

Can we at least make this a method? People don't think much of the order in which you set properties, but they do care about the order in which you invoke methods. It could be a Set-only method. (No need to "Get" it back out.)

Reply:
Yes let's make this a method.

`void SetLoaderDllFolderPath(string)`

Reply:
Okey, I have make a PR to do this:[https://dev.azure.com/microsoft/Edge/_git/chromium.src/pullrequest/7378162](url)